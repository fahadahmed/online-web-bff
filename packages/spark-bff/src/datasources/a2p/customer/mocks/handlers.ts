import { getAdminOverviewMock } from './getAdminOverview';
import { getCustomerOverviewMock } from './getCustomerOverview';
import { getContentProvidersMock } from './getContentProviders';
import { getServiceRequestsMock } from './getServiceRequests';
import { getShortcodesMock } from './getShortcodes';
import { getShortcodeMock } from './getShortcode';
import { getUsageMock } from './getUsage';
import { addContentProviderMock } from './addContentProvider';
import { addServiceRequestMock } from './addServiceRequest';
import { modifyShortcodeMock } from './modifyShortcode';
import { modifyContentProviderMock } from './modifyContentProvider';
import { deleteContentProviderMock } from './deleteContentProvider';

const handlers = [
  addContentProviderMock,
  addServiceRequestMock,
  getAdminOverviewMock,
  getCustomerOverviewMock,
  getContentProvidersMock,
  getServiceRequestsMock,
  getShortcodesMock,
  getShortcodeMock,
  getUsageMock,
  modifyShortcodeMock,
  modifyContentProviderMock,
  deleteContentProviderMock,
];

export default handlers;
