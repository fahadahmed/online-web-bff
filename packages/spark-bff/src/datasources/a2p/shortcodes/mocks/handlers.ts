import { getAdminShortcodesMock } from './getShortcodes';
import { getAdminShortcodesOverviewMock } from './getShortcodesOverview';

const handlers = [getAdminShortcodesMock, getAdminShortcodesOverviewMock];

export default handlers;
