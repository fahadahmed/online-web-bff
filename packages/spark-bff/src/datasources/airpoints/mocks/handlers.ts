import { rest } from 'msw';
import airpointsResponseMock from './airpoints-response.mock.json';
import addAirpointsResponseMock from './add-airpoints-response.mock.json';
import deleteAirpointsResponseMock from './delete-airpoints-response.mock.json';
import airpointsInactiveResponseMock from './airpoints-inactive-response.mock.json';
import addAirpointsErrorResponseMock from './add-airpoints-error-response.mock.json';
import deleteAirpointsErrorResponseMock from './delete-airpoints-error-response.mock.json';

interface AddAirpointsMockBody {
  accountNumber?: string;
  airpointsNumber?: string;
  firstName?: string;
  lastName?: string;
  status?: string;
}

export const airpointsResponseMocks = rest.get<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber === '123456789') {
      /* Simulate error responses */
      return res(ctx.status(200), ctx.json(airpointsResponseMock));
    }
    return res(ctx.status(200), ctx.json(airpointsInactiveResponseMock));
  },
);

export const updateAirpointsResponseMocks = rest.post<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async (
    { params: { accountNumber }, body: { airpointsNumber } },
    res,
    ctx,
  ) => {
    /* add airpoints */
    if (airpointsNumber) {
      if (airpointsNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(400), ctx.json(addAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(addAirpointsResponseMock));
    }
    /* delete airpoints */
    if (accountNumber) {
      if (accountNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(200), ctx.json(deleteAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(deleteAirpointsResponseMock));
    }
    return null;
  },
);

const handlers = [airpointsResponseMocks, updateAirpointsResponseMocks];

export default handlers;
