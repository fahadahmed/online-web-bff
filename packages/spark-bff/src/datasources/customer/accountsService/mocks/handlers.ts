import { rest } from 'msw';
import accountSummaryMock from './account-summary.mock.json';
import emptyAccountListMock from './empty-account-list.mock.json';
import singleAccountListMock from './single-account-list.mock.json';
import accountListMock from './account-list.mock.json';
import guestAccountListMock from './guest-account-list.mock.json';
import unauthorizedAccountListMock from './unauthorized-account-list.mock.json';

const baseUrl = 'http://testhost/v1/customer/account';

export const accountSummaryHandler = rest.get<unknown, { accountId: string }>(
  `${baseUrl}/:accountId/summary`,
  async (req, res, ctx) => {
    const { accountId } = req.params;
    if (accountId.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountSummaryMock));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

export const accountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(accountListMock));
  },
);

export const singleAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(singleAccountListMock));
  },
);

export const emptyAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(emptyAccountListMock));
  },
);

export const unauthorizedAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(401), ctx.json(unauthorizedAccountListMock));
  },
);

export const guestAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(403), ctx.json(guestAccountListMock));
  },
);

const handlers = [accountSummaryHandler, accountListHandler];

export default handlers;
