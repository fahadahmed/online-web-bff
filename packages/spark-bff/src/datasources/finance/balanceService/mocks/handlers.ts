import { rest } from 'msw';
import accountBalanceDetails from './account-balance.mock.json';

const baseUrl = 'http://testhost/v1/finance/balance/';

export const accountBalanceMock = rest.get<unknown, { accountNumber: string }>(
  `${baseUrl}account/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountBalanceDetails));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

const handlers = [accountBalanceMock];

export default handlers;
