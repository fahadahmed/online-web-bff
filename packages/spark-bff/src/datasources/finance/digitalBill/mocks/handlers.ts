import { rest } from 'msw';
import accountBillingSummary from './account-billing-summary.mock.json';
import accountTransactions from './account-transactions.mock.json';

const baseUrl = 'http://testhost/v2/finance';

export const accountBillingSummaryMock = rest.get(
  `${baseUrl}/:accountNumber/bills/summary`,
  async (_, res, ctx) => {
    return res(ctx.json(accountBillingSummary));
  },
);

export const accountTransactionsMock = rest.get(
  `${baseUrl}/:accountNumber/bills/transactions`,
  async (_, res, ctx) => {
    return res(ctx.json(accountTransactions));
  },
);

const handlers = [accountBillingSummaryMock, accountTransactionsMock];

export default handlers;
