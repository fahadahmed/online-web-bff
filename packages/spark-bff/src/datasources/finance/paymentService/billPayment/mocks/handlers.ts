import { rest } from 'msw';
import poliPayBillSuccess from './poliPayBill-success.mock.json';
import poliPayBillValidationFailure from './poliPayBill-failure.mock.json';
import creditCardPayBillSuccess from './creditCard-payBill-success.mock.json';
import creditCardPayBillValidationFailure from './creditCard-payBill-failure.mock.json';
import deleteAutoPayBillSuccess from './delete-autoPayBill-success.mock.json';
import deleteAutoPayBillFailure from './delete-autoPayBill-failure.mock.json';

const baseUrl = 'http://testhost/v1/finance/payment';

export const addPoliPayBillMock = rest.post<unknown, { accountNumber: string }>(
  `${baseUrl}/polipay/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if ((accountNumber as string).match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(poliPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(poliPayBillSuccess));
  },
);

export const addCreditCardPayBillMock = rest.post<
  unknown,
  { accountNumber: string }
>(
  `${baseUrl}/creditcard/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if (accountNumber.match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(creditCardPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(creditCardPayBillSuccess));
  },
);

export const deleteAutoPayBillMock = rest.delete<
  unknown,
  { accountNumber: string; paymentMethodId: string }
>(
  `${baseUrl}/bill/:accountNumber/paymentsetup/:paymentMethodId`,
  async ({ params: { accountNumber, paymentMethodId } }, res, ctx) => {
    /** If accountNumber does not match the given pattern,
     for example, 746738998.
     Or If paymentMethodId does not match the given pattern,
     for example,"1-S0123-23"
     it returns failure response, "Request Validation Failure".
     */
    if (
      accountNumber.match(/^\d+$/) === null ||
      paymentMethodId.match(/^[\dA-Z][\dA-Z-]+[\dA-Z]$/) === null
    ) {
      return res(ctx.status(400), ctx.json(deleteAutoPayBillFailure));
    }
    return res(ctx.status(200), ctx.json(deleteAutoPayBillSuccess));
  },
);

const handlers = [
  addCreditCardPayBillMock,
  addPoliPayBillMock,
  deleteAutoPayBillMock,
];

export default handlers;
