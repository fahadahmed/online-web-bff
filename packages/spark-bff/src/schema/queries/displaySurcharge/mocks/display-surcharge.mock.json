{
  "PAY_BILL": {
    "operationType": "PAY_BILL",
    "masterCard": "3%",
    "visa": "3%",
    "amex": "2%",
    "dinersClub": "2%",
    "displayText": [
      "American Express and Diners Club incur a 2% surcharge.",
      "Visa/MasterCard, Prezzy Card, international credit, debit or charge cards incur a 3% surcharge."
    ]
  },
  "SETUP_MONTHLY_PAYMENT": {
    "operationType": "SETUP_MONTHLY_PAYMENT",
    "masterCard": "3%",
    "visa": "3%",
    "amex": "2%",
    "dinersClub": "2%",
    "displayText": [
      "Automated payments using a credit/debit card incurs a 1.5% surcharge (plus GST, if any). Surcharge applies."
    ]
  }
}
