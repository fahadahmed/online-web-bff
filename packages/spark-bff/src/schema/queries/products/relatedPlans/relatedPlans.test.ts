import { createTestClient, gql } from 'test/apolloTestUtils';
import { server as mockServer } from 'test/server';
import { relatedPlansHandler } from 'datasources/product/offers/mocks/handlers';

beforeEach(() => {
  mockServer.use(relatedPlansHandler);
});

test('returns success for fetching the related plans based on categoryId and offeringId', async () => {
  const { query } = createTestClient();

  const res = await query({
    query: gql`
      query {
        relatedPlans(
          input: {
            categoryId: "vas_addon"
            offerIds: ["spotify_premium-of_jystv"]
            channel: personalshop
          }
        ) {
          bundles {
            id
            offerDetails {
              id
              name
              description
              price {
                basePrice
                frequency {
                  period
                  value
                }
                priceType
              }
              images {
                thumbnailUrl
              }
              entitlements {
                name
                iconUrl
                longDescription
                type
              }
              promotions {
                name
                productType
                shortDescription
                longDescription
                iconUrl
              }
            }
          }
        }
      }
    `,
  });
  expect(res.errors).toBeUndefined();
  expect(res.data).toMatchSnapshot();
});
