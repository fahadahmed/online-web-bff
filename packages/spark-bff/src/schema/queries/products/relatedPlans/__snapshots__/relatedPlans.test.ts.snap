// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns success for fetching the related plans based on categoryId and offeringId 1`] = `
Object {
  "relatedPlans": Object {
    "bundles": Array [
      Object {
        "id": "d18a3d99-2cb7-47a3-a295-e10b6bc298ca",
        "offerDetails": Array [
          Object {
            "description": "$79.99 Endless Mobile Plan",
            "entitlements": Array [
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": null,
                "name": "Roaming data",
                "type": "Roaming",
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "Unlimited NZ Landline and On Net Mins",
                "type": "Voice",
              },
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": "Text your friends, its unlimited!!",
                "name": "Unlimited texts (NZ only)",
                "type": "SMS",
              },
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": null,
                "name": "Endless data (40GB)",
                "type": "Data",
              },
            ],
            "id": "dollar79_99_endless_mobile_plan",
            "images": Array [],
            "name": "$79.99 Endless Mobile Plan",
            "price": Object {
              "basePrice": 79.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
              "priceType": "Recurring",
            },
            "promotions": Array [
              Object {
                "iconUrl": null,
                "longDescription": null,
                "name": "Eligible for group discount",
                "productType": "MARKETING",
                "shortDescription": "",
              },
              Object {
                "iconUrl": "https://cdn-icons-png.flaticon.com/512/2111/2111624.png",
                "longDescription": "Browse and stream while enjoying Spotify on us!",
                "name": "Included Spotify Premium",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Browse and stream while enjoying Spotify on us!",
              },
              Object {
                "iconUrl": null,
                "longDescription": "Visual Voicemail allows you to hear and read your voicemail messages.  You can still use the traditional voice prompted voicemail. Any change you make in the app, is made in the traditional voicemail.",
                "name": "Visual Voicemail",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Visual Voicemail",
              },
              Object {
                "iconUrl": null,
                "longDescription": null,
                "name": "Save up to 35% on multiple plans",
                "productType": "DEAL",
                "shortDescription": "",
              },
            ],
          },
          Object {
            "description": "$99.99 Endless Group Plan",
            "entitlements": Array [
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": null,
                "name": "Roaming data",
                "type": "Roaming",
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "Unlimited NZ Landline and On Net Mins",
                "type": "Voice",
              },
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": "Text your friends, its unlimited!!",
                "name": "Unlimited texts (NZ only)",
                "type": "SMS",
              },
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": null,
                "name": "Endless data (100GB)",
                "type": "Data",
              },
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": null,
                "name": "Total Data Used",
                "type": "Data",
              },
            ],
            "id": "dollar99_99_endless_group_plan",
            "images": Array [],
            "name": "$99.99 Endless Group Plan",
            "price": Object {
              "basePrice": 99.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
              "priceType": "Recurring",
            },
            "promotions": Array [
              Object {
                "iconUrl": "https://cdn-icons-png.flaticon.com/512/2111/2111624.png",
                "longDescription": "Browse and stream while enjoying Spotify on us!",
                "name": "Included Spotify Premium",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Browse and stream while enjoying Spotify on us!",
              },
              Object {
                "iconUrl": null,
                "longDescription": "Visual Voicemail allows you to hear and read your voicemail messages.  You can still use the traditional voice prompted voicemail. Any change you make in the app, is made in the traditional voicemail.",
                "name": "Visual Voicemail",
                "productType": "DEAL",
                "shortDescription": "Visual Voicemail",
              },
              Object {
                "iconUrl": null,
                "longDescription": null,
                "name": "Eligible for group discount",
                "productType": "MARKETING",
                "shortDescription": "",
              },
              Object {
                "iconUrl": null,
                "longDescription": null,
                "name": "Save up to 35% on multiple plans",
                "productType": "DEAL",
                "shortDescription": "",
              },
            ],
          },
          Object {
            "description": "$59.99 Endless Mobile Plan",
            "entitlements": Array [
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": null,
                "name": "Roaming data",
                "type": "Roaming",
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "Unlimited NZ Landline and On Net Mins",
                "type": "Voice",
              },
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": "Text your friends, its unlimited!!",
                "name": "Unlimited texts (NZ only)",
                "type": "SMS",
              },
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": null,
                "name": "Endless data (10GB)",
                "type": "Data",
              },
            ],
            "id": "dollar59_99_endless_mobile_plan",
            "images": Array [],
            "name": "$59.99 Endless Mobile Plan",
            "price": Object {
              "basePrice": 59.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
              "priceType": "Recurring",
            },
            "promotions": Array [
              Object {
                "iconUrl": "https://cdn-icons-png.flaticon.com/512/2111/2111624.png",
                "longDescription": "If you're on an eligible mobile plan that gives you 50% off Spotify Premium, or if you want to buy Spotify Premium for $14.99 per month, go to My Mobile > buy Extras or pack. Spotify Premium is included in selected mobile plans. To find out more, select View Plans.",
                "name": "Half price Spotify Premium",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Enjoy non-stop music with Spotify Premium so you can access millions of songs ad-free and on-demand",
              },
              Object {
                "iconUrl": null,
                "longDescription": "Visual Voicemail allows you to hear and read your voicemail messages.  You can still use the traditional voice prompted voicemail. Any change you make in the app, is made in the traditional voicemail.",
                "name": "Visual Voicemail",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Visual Voicemail",
              },
            ],
          },
          Object {
            "description": "$39.99 3GB Rollover",
            "entitlements": Array [
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": null,
                "name": "Roaming data",
                "type": "Roaming",
              },
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": null,
                "name": "Overage Data",
                "type": "Data",
              },
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": "Really great for video streaming",
                "name": "3GB Rollover data",
                "type": "Data",
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "Rollover Voice",
                "type": "Voice",
              },
              Object {
                "iconUrl": "unlimited_small",
                "longDescription": null,
                "name": "Rollover Data",
                "type": "Rollover Data",
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": "Chat for a really long time",
                "name": "300 Rollover NZ mins",
                "type": "Voice",
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": "Text your friends, its unlimited!!",
                "name": "Unlimited texts (NZ only)",
                "type": "SMS",
              },
            ],
            "id": "dollar_39_endless_mobile_plan-of_luqnb",
            "images": Array [],
            "name": "$39.99 3GB Rollover",
            "price": Object {
              "basePrice": 39.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
              "priceType": "Recurring",
            },
            "promotions": Array [
              Object {
                "iconUrl": "https://cdn-icons-png.flaticon.com/512/2111/2111624.png",
                "longDescription": "If you're on an eligible mobile plan that gives you 50% off Spotify Premium, or if you want to buy Spotify Premium for $14.99 per month, go to My Mobile > buy Extras or pack. Spotify Premium is included in selected mobile plans. To find out more, select View Plans.",
                "name": "Half price Spotify Premium",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Enjoy non-stop music with Spotify Premium so you can access millions of songs ad-free and on-demand",
              },
              Object {
                "iconUrl": null,
                "longDescription": "Visual Voicemail allows you to hear and read your voicemail messages.  You can still use the traditional voice prompted voicemail. Any change you make in the app, is made in the traditional voicemail.",
                "name": "Visual Voicemail",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Visual Voicemail",
              },
            ],
          },
          Object {
            "description": "Smart Plan",
            "entitlements": Array [],
            "id": "wearableplan_smart-of_oh42p",
            "images": Array [],
            "name": "Smart Plan",
            "price": Object {
              "basePrice": 7.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
              "priceType": "Recurring",
            },
            "promotions": Array [],
          },
          Object {
            "description": "$12.99 One Number Wearable Plan",
            "entitlements": Array [],
            "id": "one_number_wearbale_plan",
            "images": Array [],
            "name": "$12.99 One Number Wearable Plan",
            "price": Object {
              "basePrice": 12.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
              "priceType": "Recurring",
            },
            "promotions": Array [],
          },
          Object {
            "description": "$14.99 Endless Wearable Plan",
            "entitlements": Array [],
            "id": "endless_wearable_plan",
            "images": Array [],
            "name": "$14.99 Endless Wearable Plan",
            "price": Object {
              "basePrice": 14.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
              "priceType": "Recurring",
            },
            "promotions": Array [],
          },
          Object {
            "description": "$25 Rollover Plan(Gold Talker)",
            "entitlements": Array [
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "Roaming data",
                "type": null,
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "500 NZ Text",
                "type": null,
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "Unlimited NZ Landline and On Net Mins",
                "type": null,
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "Rollover Data",
                "type": null,
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "500MB NZ Rollover Data",
                "type": null,
              },
            ],
            "id": "gold_talker",
            "images": Array [],
            "name": "$25 Rollover Plan",
            "price": Object {
              "basePrice": 25,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
              "priceType": "Recurring",
            },
            "promotions": Array [
              Object {
                "iconUrl": null,
                "longDescription": null,
                "name": "Eligible for group discount",
                "productType": "MARKETING",
                "shortDescription": "",
              },
              Object {
                "iconUrl": null,
                "longDescription": "\\"Visual Voicemail allows you to hear and read your voicemail messages.  You can still use the traditional voice prompted voicemail. Any change you make in the app, is made in the traditional voicemail.\\"",
                "name": "Visual Voicemail",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Visual Voicemail",
              },
            ],
          },
          Object {
            "description": "$15 Rollover Plan(Gold Basic)",
            "entitlements": Array [
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "100 On net minutes",
                "type": null,
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "100MB NZ Rollover Data",
                "type": null,
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "Roaming data",
                "type": null,
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": "",
                "name": "Unlimited NZ Landline and On Net Mins",
                "type": null,
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "100 SMS",
                "type": null,
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "Rollover Data",
                "type": null,
              },
              Object {
                "iconUrl": "check_small",
                "longDescription": null,
                "name": "Rollover Voice",
                "type": null,
              },
            ],
            "id": "gold_basic",
            "images": Array [],
            "name": "$15 Rollover Plan",
            "price": Object {
              "basePrice": 15,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
              "priceType": "Recurring",
            },
            "promotions": Array [
              Object {
                "iconUrl": null,
                "longDescription": "Visual Voicemail allows you to hear and read your voicemail messages.  You can still use the traditional voice prompted voicemail. Any change you make in the app, is made in the traditional voicemail.",
                "name": "Visual Voicemail",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Visual Voicemail",
              },
            ],
          },
        ],
      },
    ],
  },
}
`;
